operator/controller/src/main/java/io/apicurio/registry/operator/util/IngressUtil.java

```diff
@@ -0,0 +1,60 @@
+package io.apicurio.registry.operator.util;
+
+import io.apicurio.registry.operator.OperatorException;
+import io.apicurio.registry.operator.api.v1.ApicurioRegistry3;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.networking.v1.HTTPIngressPath;
+import io.fabric8.kubernetes.api.model.networking.v1.Ingress;
+import io.fabric8.kubernetes.api.model.networking.v1.IngressRule;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.function.Consumer;
+
+import static io.apicurio.registry.operator.resource.ResourceFactory.COMPONENT_APP;
+import static io.apicurio.registry.operator.resource.ResourceFactory.COMPONENT_UI;
+import static io.apicurio.registry.operator.util.Util.isBlank;
+
+public class IngressUtil {
+
+    private static final Logger log = LoggerFactory.getLogger(IngressUtil.class);
+
+    private IngressUtil() {
+    }
+
+    public static String getHost(String component, ApicurioRegistry3 p) {
+        String host = null;
+        if (COMPONENT_APP.equals(component)) {
```

Comment:
I'm not convinced about this refactoring to a `Util` class.
The fact that we need to match on strings to customize the logic is a strong indication that this is not the right level of abstraction.
The rule of thumb for "Util", "Common" and similar classes is:
> if you can't name it properly it should probably not exists

I propose that we remove this `IngressUtil` class, if the cost is some code duplication, I'm ok with it for now.

Reply:
We can have a separate session later to discuss improving the code structure.

Reply:
ok, deal